### PR TITLE
try fixing for new pyjulia

### PR DESCRIPTION
**Pull request by rejuvyesh**

### CODE REVIEW

**Comment by rejuvyesh** [comment]
@zsunberg Let's discuss here in terms of what is required. Currently trying to use `stepthrough` from POMDPSimulators fails from the python side. Rest of it works! I'll remove the automatic install of QMDP.jl but it's useful for testing for now.

**Comment by zsunberg** [comment]
We should get something like #5 working. Note that QuickPOMDPs has two interfaces (see the QuickPOMDPs README). The current tiger.py uses the old interface; the new lightdark.py that I added in #5 uses the newer more powerful interface that uses distributions

**Comment by rejuvyesh** [comment]
```
*** RuntimeError: <PyCall.jlwrap (in a Julia function called from Python)
JULIA: UndefRefError: access to undefined reference
Stacktrace:
 [1] getproperty(x::Type, f::Symbol)
   @ Base ./Base.jl:28
 [2] #s2620#1
   @ ~/.julia/packages/Tricks/TyBuj/src/Tricks.jl:20 [inlined]
 [3] var"#s2620#1"(T::Any, ::Any, f::Any, t::Any)
   @ Tricks ./none:0
 [4] (::Core.GeneratedFunctionStub)(::Any, ::Vararg{Any, N} where N)
   @ Core ./boot.jl:571
 [5] reward(::QuickPOMDPs.QuickPOMDP{UUID("3f4dc2e9-2f1d-48b7-8536-b7117279eeef"), Int64, Int64, Float64, NamedTuple{(:stateindex, :isterminal, :states, :discount, :actions, :obstype, :observation, :actionindex, :transition, :reward, :initialstate), Tuple{Dict{Int64, Int64}, PyCall.var"#fn#26"{PyCall.var"#fn#25#27"{PyObject}}, UnitRange{Int64}, Float64, Vector{Int64}, DataType, PyCall.var"#fn#26"{PyCall.var"#fn#25#27"{PyObject}}, Dict{Int64, Int64}, PyCall.var"#fn#26"{PyCall.var"#fn#25#27"{PyObject}}, PyCall.var"#fn#26"{PyCall.var"#fn#25#27"{PyObject}}, POMDPModelTools.Uniform{Set{Int64}}}}}, ::Int64, ::Int64, ::Int64)
   @ QuickPOMDPs ~/.julia/dev/QuickPOMDPs/src/quick.jl:241
 [6] invokelatest(::Any, ::Any, ::Vararg{Any, N} where N; kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
   @ Base ./essentials.jl:708
 [7] invokelatest(::Any, ::Any, ::Vararg{Any, N} where N)
   @ Base ./essentials.jl:706
 [8] _pyjlwrap_call(f::Function, args_::Ptr{PyCall.PyObject_struct}, kw_::Ptr{PyCall.PyObject_struct})
   @ PyCall ~/.julia/packages/PyCall/BD546/src/callback.jl:28
 [9] pyjlwrap_call(self_::Ptr{PyCall.PyObject_struct}, args_::Ptr{PyCall.PyObject_struct}, kw_::Ptr{PyCall.PyObject_struct})
   @ PyCall ~/.julia/packages/PyCall/BD546/src/callback.jl:44>
   
```
It seems like `static_hasmethod` doesn't work here from python because of `invokelatest`?

**Comment by zsunberg** [comment]
Yes, exactly. That's why we need to wrap the python function in a Julia
function in QuickPOMDPs.preprocess

(Sent from phone)

**Comment by zsunberg** [comment]
Or figure out another workaround for my static_hasmethod hack

(Sent from phone)

**Comment by zsunberg** [comment]
Nice @rejuvyesh I was going to work on lightdark, but I see you have already started. Let me know if you get stuck or need help.

File: quickpomdps/__init__.py [hunk 1]
```diff
@@ -1,6 +1,9 @@
 import os
 
+
 import julia
+from julia.api import Julia
+jl = Julia(compiled_modules=False)
```
Comment (zsunberg) [inline quickpomdps/__init__.py:6]:
```suggestion
```
Looks like the tests pass without this.

**Comment by zsunberg** [comment]
The `preprocess` part is going to be harder than I thought because we have to convert the PyObject to a Julia `Function` with the correct number of arguments.

We can introspect with `inspect.signature` in Python, but it will still be tricky to get every edge case right. I added the ability to have a different `preprocess` for every argument: https://github.com/JuliaPOMDP/QuickPOMDPs.jl/commit/e7be380ed700b958cf31a6243d9e95302023db79 . That should make things easier.

File: quickpomdps/__init__.py (hunk 1 above)
Comment (rejuvyesh) [reply quickpomdps/__init__.py:6]: Yeah, I needed this to work with anaconda python. @lassepe what python installation are you using or being used in CI?

Comment (lassepe) [reply quickpomdps/__init__.py:6]: I'm using the [Python asdf plugin](https://github.com/danhper/asdf-python) which uses [python-build](https://github.com/pyenv/pyenv/tree/master/plugins/python-build) under the hood. The CI uses the setup python action: https://github.com/actions/setup-python.

That said, it also works with my system python. Only important thing is to have libpython as shared object available.

Comment (zsunberg) [reply quickpomdps/__init__.py:6]: @lassepe is the system python that you are saying works on Arch?

Comment (lassepe) [reply quickpomdps/__init__.py:6]: Yes

**Comment by zsunberg** [comment]
I've been working on this, and am almost done in #5!

**Pull request closed without merging**